src/index.js

```diff
@@ -254,12 +258,12 @@ class KadDHT extends EventEmitter {
    *
    * @returns {Promise<void>}
    */
-  stop () {
+  async stop () {
     this._running = false
     this.randomWalk.stop()
-    this.providers.stop()
+    await this.network.stop()
```

Comment:
This does not need to be a promise, but `network.stop` and `network.start` need to be fixed. Registrar does not return a promise anymore: https://github.com/libp2p/js-libp2p/blob/master/src/registrar.js#L87

Reply:
I've removed all of the unnecessary async.